src/taxinyc/data_processor.py

```
@@ -0,0 +1,49 @@
+            ('onehot', OneHotEncoder(handle_unknown='ignore'))
+        ])
+        
+        # Combine preprocessing steps
+        self.preprocessor = ColumnTransformer(
+            transformers=[
+                ('num', numeric_transformer, self.config['num_features']),
+                ('cat', categorical_transformer, self.config['cat_features'])
+            ])
+
+    def split_data(self, test_size=0.2, random_state=42):
+        return train_test_split(self.X, self.y, test_size=test_size, random_state=random_state)
```

Ruff

No newline at end of file

Add trailing newline

(W292)